src/main/java/uk/gov/hmcts/reform/hmc/api/services/HearingsServiceImpl.java

```diff
@@ -156,4 +162,104 @@ private void integrateVenueDetails(Hearings caseHearingsResponse) {
             caseHearingsResponse.setCaseHearings(caseHearings);
         }
     }
+
+    /**
+     * This method will fetch all the hearings which belongs to a particular caseRefNumber.
+     *
+     * @param caseReference CaseRefNumber to take all the hearings belongs to this case.
+     * @param authorization authorization header.
+     * @param serviceAuthorization serviceAuthorization header
+     * @return hearingDetails, all the hearings which belongs to a particular caseRefNumber.
+     */
+    @Override
+    public Hearings getHearingsByCaseId(
+            String caseReference, String authorization, String serviceAuthorization) {
+
+        final String userToken = idamTokenGenerator.generateIdamTokenForHearingCftData();
+        final String s2sToken = authTokenGenerator.generate();
+
+        try {
+            hearingDetails = hearingApiClient.getHearingDetails(userToken, s2sToken, caseReference);
+            integrateVenueDetailsForCaseId(hearingDetails);
+        } catch (HttpClientErrorException | HttpServerErrorException exception) {
+            log.info("Hearing api call HttpClientError exception {}", exception.getMessage());
+        } catch (FeignException exception) {
+            log.info("Hearing api call Feign exception {}", exception.getMessage());
+        } catch (Exception exception) {
+            log.info("Hearing api call Exception exception {}", exception.getMessage());
+        }
+
+        return hearingDetails;
+    }
+
+    /**
+     * This method will fetch all the hearings which belongs to a particular caseRefNumber.
+     *
+     * @param caseIds caseId list to take all the hearings belongs to each case.
+     * @param authorization authorization header.
+     * @param serviceAuthorization serviceAuthorization header
+     * @return casesWithHearings, List of cases with all the hearings which belongs to all caseIds
+     *     passed.
+     */
+    @Override
+    public List<Hearings> getHearingsByListOfCaseIds(
+            List<String> caseIds, String authorization, String serviceAuthorization) {
+
+        List<Hearings> casesWithHearings = new ArrayList<>();
+        if (!caseIds.isEmpty()) {
+            final String userToken = idamTokenGenerator.generateIdamTokenForHearingCftData();
+            final String s2sToken = authTokenGenerator.generate();
+
+            for (String caseId : caseIds) {
+                try {
+                    hearingDetails =
+                            hearingApiClient.getHearingDetails(userToken, s2sToken, caseId);
+                    integrateVenueDetailsForCaseId(hearingDetails);
```

Comment:
Venue Details can be attached outside the forloop. 
Also, we need to call below reference data api only once. And then tag the venues against the venue ids which we receive the venue details, we need to map it against Venue ID retrieved from hearings api.

http://rd-location-ref-api-aat.service.core-compute-aat.internal/refdata/location/court-venues/services?service_code=ABA5

You can see this is implemented in prl-cos-api
https://github.com/hmcts/prl-cos-api/blob/master/src/main/java/uk/gov/hmcts/reform/prl/services/LocationRefDataService.java